### DISCUSSION TITLE
Actor cache invalidation

### DISCUSSION THREAD

**Discussion by monteeya** [Automattic/wordpress-activitypub#360] [Q&A]
Hi,

I'm using a Mastodon account to test this plugin. It seems that actor info is aggressively cached on their side, is there any way to invalidate that cache?

**Comment by mediaformat** [comment]
Normally Mastodon will re-fetch the actor profile after receiving a new post.

**Reply by monteeya** [reply]
I've seen that you mention that in another issue here, but it doesn't work in my case. I noticed that if I change my Mastodon profile, the changes are immediately updated for followers, so I guess there has to be some way to notify it?

**Reply by pfefferle** [reply]
Also on other instances? How have you proofed that?

**Reply by monteeya** [reply]
Yes, it was just a quick test with accounts on two Mastodon servers, but now I tried to reproduce it with two different servers, and the changes weren't updated (so it may have been some kind of fluke).

Is there some other way to invalidate that cache?

**Reply by pfefferle** [reply]
I am not aware of any way to invalidate or update caches using ActivityPub. I have some ideas how it should work, but I have no idea how or even if it is supported by Mastodon.

So to be sure, you have to ask the Mastodon crowd I think.

**Reply by monteeya** [reply]
OK, thank you.

> I have some ideas how it should work, but I have no idea how or even if it is supported by Mastodon.

I'm curious about this if you want to share more details.

**Reply by mediaformat** [reply]
I just tested and it worked. Have you checked your Actor json is not cached? You can use this to inspect https://json.blinry.org/#https://chaos.social/users/blinry
The other thing is that the mastodon instances you are testing on need to be following the actor.

**Reply by monteeya** [reply]
I tried it again now and it didn't work (no, actor json is not cached anywhere on my side). Maybe it depends on how the Mastodon server is configured.

**Reply by mediaformat** [reply]
On Mastodon all I could find was this https://github.com/mastodon/mastodon/issues/24306

**Reply by monteeya** [reply]
Thank you, I added a comment there, but I don't know if it'll make any difference considering that the original issue has been ignored.

I'm surprised that people are okay with this, having absolutely no control over the actor cache seems like a serious issue.

**Comment by pfefferle** [comment]
> I'm curious about this if you want to share more details.

In the end, a profile change could be a simple `Update` Activity, like for example

```JSON
{
  "@context": "https://www.w3.org/ns/activitystreams",
  "id": "https://wordpress.example/activity/update",
  "type": "Update",
  "actor": "https://wordpress.example/authors/name",
  "object": "https://wordpress.example/authors/name"
}
```

on profile changes.

**Reply by pfefferle** [reply]
Maybe we should simply implement it for the plugin and hope for adoption, to work against the hen/egg issue.

**Reply by monteeya** [reply]
I think that's a good idea, and it seems to be aligned with the ActivityPub specification. Maybe it's even supported already by some implementations.
